src/Features/VisualBasic/Portable/ConvertIfToSwitch/VisualBasicConvertIfToSwitchCodeRefactoringProvider.vb

```diff
@@ -1,197 +1,154 @@
-    Partial Friend NotInheritable Class VisualBasicConvertIfToSwitchCodeRefactoringProvider
+    Friend NotInheritable Class VisualBasicConvertIfToSwitchCodeRefactoringProvider
         Inherits AbstractConvertIfToSwitchCodeRefactoringProvider
 
+        Private Shared ReadOnly s_operatorMap As Dictionary(Of BinaryOperatorKind, (CaseClauseKind As SyntaxKind, OperatorTokenKind As SyntaxKind)) =
+            New Dictionary(Of BinaryOperatorKind, (SyntaxKind, SyntaxKind))() From
+            {
+                {BinaryOperatorKind.NotEquals, (SyntaxKind.CaseNotEqualsClause, SyntaxKind.LessThanGreaterThanToken)},
+                {BinaryOperatorKind.LessThan, (SyntaxKind.CaseLessThanClause, SyntaxKind.LessThanToken)},
+                {BinaryOperatorKind.GreaterThan, (SyntaxKind.CaseGreaterThanClause, SyntaxKind.GreaterThanToken)},
+                {BinaryOperatorKind.LessThanOrEqual, (SyntaxKind.CaseLessThanOrEqualClause, SyntaxKind.LessThanEqualsToken)},
+                {BinaryOperatorKind.GreaterThanOrEqual, (SyntaxKind.CaseGreaterThanOrEqualClause, SyntaxKind.GreaterThanEqualsToken)}
+            }
+
         <ImportingConstructor>
         Public Sub New()
         End Sub
 
-        Protected Overrides Function CreateAnalyzer(syntaxFacts As ISyntaxFactsService, semanticModel As SemanticModel) As IAnalyzer
-            Return New VisualBasicAnalyzer(syntaxFacts, semanticModel)
+        Public Overrides Function CreateAnalyzer(syntaxFacts As ISyntaxFactsService) As IAnalyzer
+            Return New VisualBasicAnalyzer(syntaxFacts)
         End Function
 
         Private NotInheritable Class VisualBasicAnalyzer
-            Inherits Analyzer(Of SyntaxList(Of StatementSyntax), ExecutableStatementSyntax, ExpressionSyntax, CaseClauseSyntax)
+            Inherits Analyzer(Of ExecutableStatementSyntax)
 
-            Public Sub New(syntaxFacts As ISyntaxFactsService, semanticModel As SemanticModel)
-                MyBase.New(syntaxFacts, semanticModel)
+            Public Sub New(syntaxFacts As ISyntaxFactsService)
+                MyBase.New(syntaxFacts)
             End Sub
 
-            Protected Overrides ReadOnly Property Title As String
-                Get
-                    Return VBFeaturesResources.Convert_to_Select_Case
-                End Get
-            End Property
+            Public Overrides Function CreateSwitchExpressionStatement(target As SyntaxNode, sections As ImmutableArray(Of SwitchSection)) As SyntaxNode
+                Throw ExceptionUtilities.Unreachable
+            End Function
 
-            Protected Overrides Function GetSwitchSectionBody(statement As SyntaxList(Of StatementSyntax)) As IEnumerable(Of SyntaxNode)
-                Return statement
+            Public Overrides Function CreateSwitchStatement(ifStatement As SyntaxNode, expression As SyntaxNode, sectionList As IEnumerable(Of SyntaxNode)) As SyntaxNode
+                Return VisualBasicSyntaxGenerator.Instance.SwitchStatement(expression, sectionList)
```

Comment:
can you comment the reason fo the difference.  i.e. why you don't just do this for C#.  my guess is that the answer is something like "in c# we have far more pieces of syntax (like tokens) that we want to preserve in the transformation".